packages/eslint-plugin/tests/rules/no-unsafe-return.test.ts

```diff
@@ -459,5 +484,268 @@ function example() {
         },
       ],
     },
+    {
+      code: `
+declare const value: any;
+async function foo() {
+  return value;
+}
+      `,
+      errors: [
+        {
+          messageId: 'unsafeReturn',
+          line: 4,
+          column: 3,
+          data: {
+            type: '`any`',
+          },
+        },
+      ],
+    },
+    {
+      code: `
+declare const value: Promise<any>;
+function foo() {
```

Comment:
I thought this is supposed to be valid, since it's not an `async` function, according to https://github.com/typescript-eslint/typescript-eslint/issues/8674#issuecomment-1998689040. Did that decision get changed?

Reply:
Note also that the error message for this case "Unsafe return of an `Promise<any>` typed value." is problematic, same as https://github.com/typescript-eslint/typescript-eslint/pull/8693#discussion_r1634134811

Reply:
I've fixed!, I had forgotten about those comments.